lib/cli/commands/orders/buy.ts

```diff
@@ -0,0 +1,47 @@
+import { callback, loadXudClient } from '../../command';
+import { Arguments, Argv } from 'yargs';
+import { PlaceOrderRequest } from '../../../proto/xudrpc_pb';
+
+export const command = 'buy <quantity> <pair_id> <price> [order_id]';
+
+export const describe = 'place a buy order';
+
+export const builder = (argv: Argv) => argv
+  .option('quantity', {
+    type: 'number',
+  })
+  .option('pair_id', {
+    type: 'string',
+    describe: 'the trading pair ticker id for the order',
+  })
+  .option('price', {
+    type: 'string',
+    describe: 'the price for limit orders, or "mkt"/"market" for market orders',
+  })
+  .option('order_id', {
+    type: 'string',
+    describe: 'the local order id to for this order',
+  })
+  .example('$0 buy 5 LTC/BTC .01 1337', 'place a limit order to buy 5 LTC @ 0.01 BTC with local order id 1337')
+  .example('$0 buy 3 LTC/BTC mkt', 'place a market order to buy 3 LTC for BTC')
+  .example('$0 buy 10 ZRX/GNT market', 'place a market order to buy 10 ZRX for GNT');
+
+export const handler = (argv: Arguments) => {
+  const request = new PlaceOrderRequest();
+
+  const numericPrice = Number(argv.price);
+  const priceStr = argv.price.toLowerCase();
+  if (isNaN(numericPrice)) {
```

Comment:
The logic in `sell.ts` is the exact same as in here. Maybe you could unify both.

Reply:
Yeah there's a lot of duplication or near duplication between these two files but I couldn't come up with a clean way to reuse the logic. After looking at this again though I think I came up with a good solution to minimize repetition. Take a look and see what you think.